toucan_connectors/common.py

```diff
@@ -1,13 +1,97 @@
 import ast
 import re
-from abc import ABC, ABCMeta, abstractmethod
 from copy import deepcopy
+from enum import Enum
 
 import pyjq
 from jinja2 import Environment, StrictUndefined, Template, meta
 from pydantic import Field
 from toucan_data_sdk.utils.helpers import slugify
 
+
+# Permissions conditions operators
+class ConditionOperator(Enum):
+    EQUAL = 'eq'
+    NOT_EQUAL = 'ne'
+    LOWER_THAN = 'lt'
+    LOWER_THAN_EQUAL = 'le'
+    GREATER_THAN = 'gt'
+    GREATER_THAN_EQUAL = 'ge'
+    IN = 'in'
+    NOT_IN = 'nin'
+    MATCHES = 'matches'
+    NOT_MATCHES = 'notmatches'
+    IS_NULL = 'isnull'
+    IS_NOT_NULL = 'notnull'
+
+    @classmethod
+    def has_value(cls, value: str) -> bool:
+        return any(value == item.value for item in cls)
+
+
+class ConditionTranslator:
+    """
+    Base class for any translator from conditions to query clause.
+
+    The `translate` method take the condition and returnthe corresponding part of a query.
+
+    Each available operator supported must be implemented in child classes.
+    """
+
+    def translate(cls, condition):
+        raise NotImplementedError
+
+    @classmethod
+    def EQUAL(cls):
```

Comment:
I don't understand why not just do something like this
```python
class ConditionTranslator(ABC):
    @classmethod
    @abstractmethod
    def translate(cls, condition):
         "main method..."

    @classmethod
    @abstractmethod
    def eq(cls, column, value) -> Any:  # return type depends on translators(string for pandas, dict for mongo, ...)
        ...

    def ...
 ```

I think it's way clearer and explicit. What do you think?